### PR TITLE
Use inheritance for TypeDefinitions and Unknowns

### PR DESCRIPTION
**Pull request by t-sommer**
see #561

### CODE REVIEW

**Comment by andreas-junghanns** [comment]
As you promised: the XML shrinks signigicantly AND readabilty is better. How can we not like this!

The example already had an issue with the type here Modelica.SIunits.AngularVelocity (Float64 or Int32). You changes kept the issue intact, maybe we can fix this here right now.

**Comment by t-sommer** [comment]
> The example already had an issue with the type here Modelica.SIunits.AngularVelocity (Float64 or Int32).

@andreas-junghanns, can you point me to the issue or suggest a change?

**Comment by otronarp** [comment]
The #561 part of this looks great and makes a lot of sense. However, to also remove the containers for `Outputs`, `Derivatives` and `InitialUnknowns` is in my opining taking optimization one step too far. If they are an ordered lists of outputs it makes more sense to actually keep them separated as a list in their own container.

**Comment by andreas-junghanns** [comment]
Otto: What is the value of this additional layer? When we discussed this, we could not see one and when I look at the examples, I like the simplicity and the removal of the strange "unknown" element.

**Comment by otronarp** [comment]
@andreas-junghanns it enforces that the lists are kept together and makes it more clear that there are actually four separate ordered lists and where these lists starts and ends, which I find nice. With the proposal you can for example have the list of outputs interleaved with the derivatives.

We could still ge rid of the Unknown element by using inheritance and using the singular elements (`Output`, `Derivative`, ...) in the lists.

But I don't have a strong opinion on this, I just don't see the benefit of removing those container elements.

**Comment by t-sommer** [comment]
> With the proposal you can for example have the list of outputs interleaved with the derivatives.

This is not true. The validation of this (illegal) example

```
<ModelStructure>
    <Output valueReference="805306368"/>
    <InitialUnknown valueReference="805306368"/>
    <Output valueReference="805306369"/>
    <InitialUnknown valueReference="805306369"/>
  </ModelStructure>
```

yields

```
lxml.etree.XMLSyntaxError: Element 'Output': This element is not expected. Expected is one of ( InitialUnknown, NumberOfEventIndicators
```

**Comment by otronarp** [comment]
I'm sorry I didn't realized that the order was enforced by xsd, then I guess that's not an issue then.

**Comment by pmai** [comment]
The original intent in making this element mandatory in 2.0 was, as far as I can tell, to make clear that the model structure MUST be supplied (i.e. it is not optional to supply it), it just happens that for very special circumstances (i.e. FMU without any outputs, and without any other necessary model structure information), it could be empty.

Which is different from the ModelStructure being optional (i.e. you could leave it out, even if you have outputs). While this is of course required already by the standard text, the XSD making the element required makes this slightly more obvious/in your face.

**Comment by chrbertsch** [comment]
Webmeeting (Pierre, Antoine, Karl, Torsten S., Christian B.): recommendation to merge this.

File: schema/fmi3ModelDescription.xsd [hunk 1]
```diff
@@ -141,45 +146,13 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 						<xs:group ref="fmi3Variable" maxOccurs="unbounded"/>
 					</xs:complexType>
 				</xs:element>
-				<xs:element name="ModelStructure">
+				<xs:element name="ModelStructure" minOccurs="0">
 					<xs:complexType>
 						<xs:sequence>
-							<xs:element name="Outputs" type="fmi3VariableDependency" minOccurs="0">
```
Comment (friedrichatgc) [inline schema/fmi3ModelDescription.xsd]: Please make such cleanups in a consistent way or skip such cosmetic changes!
Before this change everything was consistent and by this cosmetic change the document is now inconsistent.
The removed container elements Output**s**, Derivative**s** and InitialUnknown**s** are still referenced at many many places in the document.

**Pull request merged**